subsys/zbus/zbus.c

```diff
@@ -8,11 +8,47 @@
 #include <zephyr/sys/iterable_sections.h>
 #include <zephyr/logging/log.h>
 #include <zephyr/sys/printk.h>
+#include <zephyr/net/buf.h>
 #include <zephyr/zbus/zbus.h>
 LOG_MODULE_REGISTER(zbus, CONFIG_ZBUS_LOG_LEVEL);
 
+#if defined(CONFIG_ZBUS_MSG_SUBSCRIBER)
+
+#if defined(CONFIG_ZBUS_MSG_SUBSCRIBER_NET_BUF_DYNAMIC)
+
+NET_BUF_POOL_HEAP_DEFINE(_zbus_msg_subscribers_pool, CONFIG_ZBUS_MSG_SUBSCRIBER_NET_BUF_POOL_SIZE,
```

Comment:
What's the reasoning behind using net bufs here instead of, for example, memory slabs or similar?

Reply:
@carlescufi 
> What's the reasoning behind using net bufs here instead of, for example, memory slabs or similar?

The main reasons are: 

1. Useful abstractions like clone (with "shallow" copies), reference counting, and so forth;
2. The `net_buf` has integration with FIFOs;
3. It is already a memory pool internally;
4. `net_buf` is used in several parts of the systems (mature);
5. I am avoiding reimplementing the wheel.

Reply:
That makes a lot of sense @rodrigopex. I wonder if we should consider renaming the primitive so we decouple it from its association with networking.

Reply:
For sure, `net_buf` is not a suitable name anymore. It is meant for general use. Any data transmission inside the system can use that, mainly if it relies on FIFOs. Another advantage is the flexibility of using or not using the HEAP for data allocation. Thanks @carlescufi